mongodb/src/main/java/org/geotools/data/mongodb/MongoDBObjectFeature.java

```diff
@@ -83,8 +83,8 @@ public BoundingBox getBounds() {
 
     @Override
     public Object getDefaultGeometry() {
-        Object o = getDBOValue(mapper.getGeometryPath());
-        return o instanceof DBObject ? mapper.getGeometry((DBObject)o) : null;
+        Object o = mapper.getGeometry(featureDBO);
+        return o instanceof Geometry ? o : null;
```

Comment:
Consider the following document:
{noformat}
{   "type": "Feature",
    "geometry": {
        "type": "Point",
        "coordinates": [ 45.52, -122.681944 ]
    },
    "properties": {
        "city": "Portland",
        "year": "2014"
        "attendance": "840"
    }
}
{noformat}

Before this change, getDefaultGeometry() did the following:

* {{getDBOValue(mapper.getGeometryPath())}} returned the Mongo geometry object:
{noformat}
 {
    "type": "Point",
     "coordinates": [ 45.52, -122.681944 ]
}
{noformat}

Then, {{mapper.getGeometry((DBObject)o)}} searches that object for the key of the default geometry attribute descriptor (in this case, "geometry"}, to try and convert to a geometry object. However, since the value passed for {{o}} was the geometry object and not the document, the key "geometry" was not found, and {{getDefaultGeometry()}} returned {{null}}. (This was discovered after stepping through the code in a debugger).

My change passes the actuall document to the {{mapper.getGeometry}} method in order to correctly get the mongo geometry object and convert it to a {{Geometry}}.